samcli/local/common/runtime_template.py

```diff
@@ -138,6 +139,7 @@ def get_local_lambda_images_location(mapping, runtime):
     # ruby runtimes in descending order
     "ruby2.7",
     "ruby2.5",
+    "rust(provided.al2)",
```

Comment:
This will require changes whenever a new init template is added. For instance, other than Rust, there is also another PR which is targeting to add new templates for GraalVM, which also uses `provided.al2` for the runtime. Instead of checking runtimes full name, we can check if they are ending with  `(provided.al2)` or `(provided)`, so that we would know that is a custom runtime, and we don't need to check the full string.

Reply:
are we certain that all custom runtimes will end with  the provideded suffix?

Reply:
If you want to use certain language, then you can use that language's runtime like `node10.x` or `java8`. But if you provide a self executable then you don't need anything extra on lambda image, so you can use `provided` or `provided.al2`.